src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -0,0 +1,325 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactTestRendererFiber
+ * @preventMunge
+ * @flow
+ */
+
+'use strict';
+
+var ReactFiberReconciler = require('ReactFiberReconciler');
+var ReactGenericBatching = require('ReactGenericBatching');
+var emptyObject = require('emptyObject');
+
+import type { TestRendererOptions } from 'ReactTestMount';
+
+type ReactTestRendererJSON = {
+  type: string,
+  props: { [propName: string]: string },
+  children: null | Array<string | number | ReactTestRendererJSON>,
+  $$typeof?: any
+}
+
+let instanceCounter = 0;
+
+class TestContainer {
+  createNodeMock: Function;
+
+  constructor(createNodeMock) {
+    this.createNodeMock = createNodeMock;
+  }
+  appendChild(child) {}
+  insertBefore(beforeChild, child) {}
+  removeChild(child) {}
+  toJSON() {}
```

Comment:
Sorry, this is the root node test as I understand it: https://github.com/facebook/react/pull/8628/files#diff-417ca2789067ee4dcafab55a48aa03cbR532

```
    it('can render and update root fragments', () => {
      var Component = (props) => props.children;

      var renderer = ReactTestRenderer.create([
        <Component>Hi</Component>,
        <Component>Bye</Component>,
      ]);
      expect(renderer.toJSON()).toEqual('Hi');
      renderer.update(<div />);
      expect(renderer.toJSON()).toEqual({
        type: 'div',
        children: null,
        props: {},
      });
      renderer.update([<Component>{42}</Component>, <Component>The answer</Component>]);
      expect(renderer.toJSON()).toEqual(42);
    });
```